### ISSUE TITLE
Should we add functions like `mae`, `sse`, `mse`, `rmse` here?

### ISSUE THREAD

**Issue by singularitti** [JuliaStats/StatsBase.jl#947]
Definition: https://www.sfu.ca/sasdoc/sashtml/ets/chap30/sect19.htm

I know these functions are very easy to be implemented with `mean`, `abs2`, `sqrt`, etc. But it would be really nice to have them predefined in Julia.

**Comment by nalimilan** [comment]
Yes if we add these it will be in StatsBase for consistency with existing functions. I have transferred the issue. But note that some of the functions you mention actually already exist with another name. Do you still think some are missing?

**Comment by ajinkya-k** [comment]
@nalimilan I think one useful thing to add would be the degree of freedom corrected versions. For instance the `msd` and `rmsd` use the length of the input vectors as the denominator though sometimes we might want to divide by `n - 1` instead. Maybe we can call that the `mse` / `rmse`

**Comment by ajinkya-k** [comment]
Also I think it might be useful to add aliases `sse` for the sum of squared error `mae` for `mad`

**Comment by nalimilan** [comment]
I guess we could add a `corrected` argument to these functions, similar to what we have for `cor`. Calling these `mse` and `rmse` would be confusing as there's no special reason why these should be corrected and not `msd` and `rmsd` AFAICT.

Concerning aliases, I'm hesitant to add these as it makes the API more complex and I don't think we do that usually. Feel free to make a PR to mention the alternative names in the docstrings though, that should make it easier to find the functions.

**Comment by ajinkya-k** [comment]
You're right. My suggestions were mostly just aliases. From my experience people in statistics mostly use `rmse` instead of `rmsd` though it might be specific to the people/school I work with
